Solar_System3.rb

```diff
@@ -0,0 +1,270 @@
+      puts "\nYour planete was created!\n"
+      puts "\n=============================================\n"
+      puts "\nHere's your planet's summary:\n\n"
+      puts new_planet.planet_summary
+      puts "\n=============================================\n"
+      return new_planet
+    when "no"
+      puts "\nPlanet erased!"
+      return nil
+    end
+  end
+
+  def supp_planet(planet_name)
+    found = nil
+    deleted = nil
+    if planet_name.downcase == "all"
+      @planets.clear
+      puts "\nAll the planets in your Solar System were deleted!"
+    else
+      @planets.each do |each_planet|
+        if each_planet.name.downcase == planet_name.downcase
+          @planets.delete(each_planet)
+          found = true
+          deleted = planet_name.capitalize
+        end
+      end
+      found == true ? (puts "\nPlanet #{deleted} was deleted!") : (puts "\nPlanet #{planet_name.capitalize} is not part of the Solar System.")
+    end
+  end
+
+  def print
+    @planets.each do |each_planet|
+      puts "\n" + each_planet.planet_summary
+    end
+  end
+
+  def list_planets_names
+    planet_list = ""
+    @planets.each do |each_planet|
+      if each_planet == @planets.last
+        planet_list << "#{each_planet.name}."
+      else
+        planet_list << "#{each_planet.name}, "
+      end
+    end
+    puts planet_list
+  end
+
+  def distance_from_sun(planet_name)
+    found = nil
+    distance = nil
+    @planets.map do |each_planet|
+      if each_planet.name.downcase == planet_name.downcase
+        distance = each_planet.distance_from_the_sun.to_i
+        found = true
+      end
+    end
+    found == true ? (return distance) : (puts "\nPlanet #{planet_name.capitalize} is not part of the Solar System.")
+    return distance
+  end
+
+  def numplanets
+    @planets.size
+  end
+
+  def planets_age
+    list_age = ""
+    puts "\nHere is the local ages of the planets in the Solar System:"
+    @planets.each do |each_planet|
+      each_planet.age = (@system_age * 365.25 / each_planet.year_length.to_i / 1_000_000_000).ceil
+      list_age << "- #{each_planet.name}: #{(@system_age * 365.25 / each_planet.year_length.to_i / 1_000_000_000).ceil} billion\n"
+    end
+    puts list_age
+  end
+
+end
+
+class Planet
+  attr_reader :name, :year_length, :distance_from_the_sun, :size, :atmosphere, :temperature, :planet_summary
+  attr_writer :age
+  def initialize(name, year_length, distance_from_the_sun, size, atmosphere, temperature)
+    @name = name
+    @year_length = year_length
+    @distance_from_the_sun = distance_from_the_sun
+    @size = size
+    @atmosphere = atmosphere
+    @temperature = temperature
+    @age
+  end
+  def planet_summary
+    return "#{@name}:\n#{@name}\'s year length is #{@year_length} days. \nIt\'s #{@distance_from_sun} million of miles from the sun and its radius is #{@size} miles. \nIt #{@atmosphere == "yes" ? "has" : "doesn't have"} an atmosphere and its average temperature is #{@temperature}F.\n"
```

Comment:
small typo: your instance variable is named `distance_from_the_sun`, not `distance_from_sun`